packages/adapter-components/src/fetch/errors.ts

```diff
@@ -15,11 +15,13 @@ export class AbortFetchOnFailure extends FatalError {
   }
 }
 
-export const getInsufficientPermissionsError: fetch.FetchResourceDefinition['onError'] = {
+export const createGetInsufficientPermissionsErrorFunction: (
+  statuses?: number[],
```

Comment:
I saw you provided default value but IMO `statuses` shouldn't be optional must be provided explicitly

```suggestion
  statuses: number[],
```